### PR TITLE
Fix tsc build

### PR DESCRIPTION
**Pull request by anotherminh**
## Related Issues
⚠️  NOTE: This PR will merge into `eligrey/initial-import` and not main ⚠️ 
4 main changes:
- Updated yarn version, which fixed a lot of type errors caused by not being able to resolve external deps
- Modified build command to do `TS_NODE_FILES=true ts-node src/scripts/build` so that ts-node would pick up on the additional type declarations.
- Moved the scripts + types declarations under src so that it's easy to reason about/configure the tsconfig.json but may not be necessary?
- Updated the type of build.ts because we were returning an array and not just BuildResult

## Security Implications

_[none]_

## System Availability

_[none]_

### CODE REVIEW

File: src/scripts/build.ts [hunk 1]
```diff
@@ -82,7 +82,3 @@ export const build = async (
 
   return results;
 };
-
-if (require.main === module) {
-  build();
-}
```
Comment (eligrey) [inline src/scripts/build.ts:88]: Doesn't removing this make `yarn build` a no-op?

Comment (anotherminh) [reply src/scripts/build.ts:88]: i moved it under `src`

**Comment by anotherminh** [comment]
ugh nevermind i think there are more type issues

**Comment by anotherminh** [comment]
sigh actually the build script doesnt work with TS_NODE_FILES=true 😢

**Pull request closed without merging**